billy-france/src/main/java/com/premiumminds/billy/france/services/builders/impl/FRManualCreditNoteEntryBuilderImpl.java

```diff
@@ -77,7 +77,7 @@ protected void validateInstance() throws BillyValidationException {
                 FRGenericInvoiceEntryBuilderImpl.LOCALIZER.getString("field.quantity"));
         BillyValidator.mandatory(cn.getUnitOfMeasure(),
                 FRGenericInvoiceEntryBuilderImpl.LOCALIZER.getString("field.unit"));
-        BillyValidator.mandatory(cn.getProduct(),
+        BillyValidator.<Object>mandatory(cn.getProduct(),
```

Comment:
Porque é que aqui precisa de `<Object>`?

Reply:
Isto vem na sequência de uma correcções que o froque fez para passar de java 7 para o 8. Pelos visto o type inference está diferente e ele estava a rebentar com o cast de Product para CharSequence porque estava a inferir o mandatory incorrecto. 
O froque mandou-me esta questão do stackoverflow que parece dizer respeito ao mesmo problema.
https://stackoverflow.com/questions/59430039/why-does-the-compiler-choose-this-generic-method-with-a-class-type-parameter-whe